config-model-api/src/test/java/com/yahoo/config/application/api/DeploymentSpecTest.java

```diff
@@ -309,6 +309,34 @@ public void deploymentSpecWithBlockUpgrade() {
         assertTrue(spec.canUpgradeAt(Instant.parse("2017-09-23T10:15:30.00Z")));
     }
 
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
Not a huge fan of using this feature to catch expected exceptions as it makes it too easy to catch the wrong one and the test will pass regardless. However, in this case the method body is small enough that it doesn't matter.